doc/developer_guide/module_manager.rst

```diff
@@ -104,7 +104,7 @@ module will only depend on previously defined modules:
 Once a ``ModuleInfo`` has been obtained, its primary role is to provide
 access to the PSyIR of the ``Container`` representing the module:
```

Comment:
I'm updating this in a dedicated section below this section.
I'll also write it in a way reflecting the changes coming from the 2nd PR of this issue since there's no activation of caching in the constructor of the module manager, yet. This comes in the 2nd PR.